rustls/src/client/hs.rs

```diff
@@ -645,6 +665,48 @@ pub(super) fn process_alpn_protocol(
     Ok(())
 }
 
+pub(super) fn process_server_cert_type_extension(
+    common: &mut CommonState,
+    config: &ClientConfig,
+    server_cert_extension: Option<&CertificateType>,
+) -> Result<(), Error> {
+    let requires_server_rpk = config
+        .verifier
+        .requires_raw_public_keys();
+    let server_offers_rpk = server_cert_extension.map_or(false, |cert_type| {
+        *cert_type == CertificateType::RawPublicKey
+    });
```

Comment:
I think this could be simplified to:

```rust
    let server_offers_rpk = matches!(server_cert_extension, Some(CertificateType::RawPublicKey));
```